website/src/pages/dictionary/Schema/index.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import Table from '@icgc-argo/uikit/Table';
+import TagButton, { TAG_TYPES } from './TagButton';
+import styles from './styles.module.css';
+import Tag from '@icgc-argo/uikit/Tag';
+import CodeList from './CodeList';
+import Regex from './Regex';
+const formatFieldType = value => {
+  switch (value) {
+    case 'string':
+      return 'TEXT';
+    default:
+      return value.toUpperCase();
+  }
+};
+
+const FieldDescription = ({ name, description }) => (
+  <div className={styles.fieldDescription}>
+    <div className={styles.name}>{name}</div>
+    <div>{description}</div>
+  </div>
+);
+
+const FieldsTag = ({ fieldCount }) => (
+  <Tag className={styles.fieldsTag}>{`${fieldCount} Field${fieldCount > 1 ? 's' : ''}`}</Tag>
+);
+
+const Schema = ({ schema, key }) => {
+  const fields = schema.fields.map(field => {
+    /**
+     * need to pass in state for Cell rendering
+     * react-table rerenders everything, change shape of codelist to pass in state
+     */
+    const { restrictions: { codeList = null } = {} } = field;
+
+    if (codeList) {
+      const [isExpanded, setExpanded] = useState(false);
```

Comment:
react hooks used inside conditions can break unexpectedly: https://reactjs.org/docs/hooks-rules.html (there's a linter that helps catch this, we can look into installing it at some point). This probably isn't breaking right now because the list of field is stable, but that can change with new feature.

It's probably better to lift the `useState` call up to root of Schema and use a `{[field: string]: boolean}` or something like that instead, then we can associate inside this map :)

Reply:
https://github.com/icgc-argo/argo-docs/pull/35/commits/c81b681dca22c52e4aca60bd7f60df6e32e3eb31